src/main/java/com/google/devtools/build/lib/analysis/skylark/StarlarkAttributeTransitionProvider.java

```diff
@@ -63,12 +66,18 @@ public StarlarkDefinedConfigTransition getStarlarkDefinedConfigTransitionForTest
   }
 
   @Override
-  public SplitTransition apply(AttributeMap attributeMap) {
+  public SplitTransition create(RuleTransitionData data) {
+    AttributeMap attributeMap = data.attributes();
     Preconditions.checkArgument(attributeMap instanceof ConfiguredAttributeMapper);
     return new FunctionSplitTransition(
         starlarkDefinedConfigTransition, (ConfiguredAttributeMapper) attributeMap);
   }
 
+  @Override
```

Comment:
How is this method used? Like I mentioned above, StarlarkAttributeTransitions are always SplitTransitions according to native code (for now, I suppose this distinction could be made more clear) but might not actually map to multiple output BuildOptions in practice.

Reply:
Callers of TransitionFactory that care whether the transition is a PatchTransition or a SplitTransition can use this to distinguish. It's about the return type and the intent, not on how many actual splits are produced.

Ideally at some point we should unify these further, but I'm not working on that immediately.